### ISSUE TITLE
Berkeley DB configure check fails on some systems

### ISSUE THREAD

**Issue by qsuscs** [neomutt/neomutt#613]
`configure.ac` checks for BDB in a way that fails at least on Gentoo:

- The directory is named like `/usr/include/db5.3`, which is missed by [configure.ac#L803](https://github.com/neomutt/neomutt/blob/044cc65932922c017ecf8913f9b02b2d627d0636/configure.ac#L803)
- The version test in [configure.ac#L810-L813](https://github.com/neomutt/neomutt/blob/044cc65932922c017ecf8913f9b02b2d627d0636/configure.ac#L810-L813) fails because of Gentoo’s multilib setup:
```c
/* This file is auto-generated by multilib-build.eclass
 * as a multilib-friendly wrapper. For the original content,
 * please see the files that are #included below.
 */

#if defined(__x86_64__) /* amd64 */
#       if defined(__ILP32__) /* x32 ABI */
#               error "abi_x86_x32 not supported by the package."
#       else /* 64-bit ABI */
#               include <x86_64-pc-linux-gnu/db5.3/db.h>
#       endif
#elif defined(__i386__) /* plain x86 */
#       error "abi_x86_32 not supported by the package."
#elif defined(__mips__)
// etc.
```

**Comment by gahr** [comment]
@qsuscs Could you please check and see whether the `devel/issue-613` branch fixes the issue for you?

**Comment by qsuscs** [comment]
Yes, although the check is still a bit weird:
```
checking for BerkeleyDB in /usr/local/include//... no
checking for BerkeleyDB in /usr/local/include/db-5.3... no
checking for BerkeleyDB in /usr/local/include/db53... no
checking for BerkeleyDB in /usr/local/include/db5.3... no
checking for BerkeleyDB in /usr/local/include/db-5... no
checking for BerkeleyDB in /usr/local/include/db5... no
checking for BerkeleyDB in /usr/local/include/db-6.2... no
checking for BerkeleyDB in /usr/local/include/db62... no
checking for BerkeleyDB in /usr/local/include/db6.2... no
checking for BerkeleyDB in /usr/local/include/db-6... no
checking for BerkeleyDB in /usr/local/include/db6... no
checking for BerkeleyDB in /usr/local/include/db-4.8... no
checking for BerkeleyDB in /usr/local/include/db48... no
checking for BerkeleyDB in /usr/local/include/db4.8... no
checking for BerkeleyDB in /usr/local/include/db-4... no
checking for BerkeleyDB in /usr/local/include/db4... no
checking for BerkeleyDB in /opt/include//... no
checking for BerkeleyDB in /opt/include/db-5.3... no
checking for BerkeleyDB in /opt/include/db53... no
checking for BerkeleyDB in /opt/include/db5.3... no
checking for BerkeleyDB in /opt/include/db-5... no
checking for BerkeleyDB in /opt/include/db5... no
checking for BerkeleyDB in /opt/include/db-6.2... no
checking for BerkeleyDB in /opt/include/db62... no
checking for BerkeleyDB in /opt/include/db6.2... no
checking for BerkeleyDB in /opt/include/db-6... no
checking for BerkeleyDB in /opt/include/db6... no
checking for BerkeleyDB in /opt/include/db-4.8... no
checking for BerkeleyDB in /opt/include/db48... no
checking for BerkeleyDB in /opt/include/db4.8... no
checking for BerkeleyDB in /opt/include/db-4... no
checking for BerkeleyDB in /opt/include/db4... no
checking for BerkeleyDB in /usr/local/include//... no
checking for BerkeleyDB in /usr/local/include/db-5.3... no
checking for BerkeleyDB in /usr/local/include/db53... no
checking for BerkeleyDB in /usr/local/include/db5.3... no
checking for BerkeleyDB in /usr/local/include/db-5... no
checking for BerkeleyDB in /usr/local/include/db5... no
checking for BerkeleyDB in /usr/local/include/db-6.2... no
checking for BerkeleyDB in /usr/local/include/db62... no
checking for BerkeleyDB in /usr/local/include/db6.2... no
checking for BerkeleyDB in /usr/local/include/db-6... no
checking for BerkeleyDB in /usr/local/include/db6... no
checking for BerkeleyDB in /usr/local/include/db-4.8... no
checking for BerkeleyDB in /usr/local/include/db48... no
checking for BerkeleyDB in /usr/local/include/db4.8... no
checking for BerkeleyDB in /usr/local/include/db-4... no
checking for BerkeleyDB in /usr/local/include/db4... no
checking for BerkeleyDB in /usr/include//... yes
checking for db_env_create in -ldb-6.0... yes
```
As you can see, it first tries quite a lot of directories and only at the very end `/usr/include`, and it also prefers 6.0 over 5.

By the way, you should note that >=6.0.20 is AGPL licensed (I don’t think this should be a problem for most people, though).

**Comment by gahr** [comment]
It looks like you have a db6 header in `/usr/include`, in addition to the db5 one you mentioned in `/usr/include/db5.3`.
If this is the case, the output you posted is working as designed.

**Comment by gahr** [comment]
... modulo the list of paths could be massaged to avoid duplicates because of `$prefix`.

**Comment by flatcap** [comment]
@qsuscs, three quick questions (that I should have asked a week ago):
- does it work?
- is it good enough (for now)?
- should I merge it?

**Comment by qsuscs** [comment]
Work, yes, but the output is still as lengthy as in my comment above.

The final binary links against `libdb-6.0.so` (it prefers 6.0 over 5.3, the only ones installed here).  It looks good enough for me (well, it doesn’t seem to be worse than before), whether to merge is up to you.

**Comment by flatcap** [comment]
Closed by 0fc7fa0
